### PR TITLE
feat: sideNav

### PR DESCRIPTION
**Pull request by LucasMaupin**
Screenshot 2024-10-03 at 14 31 29
Screenshot 2024-10-03 at 14 31 14

### CODE REVIEW

File: src/components/sideNav/SideNav.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  // Setting open to false will retract the sideNav menu, setting a smaller width, and all the adjustments on the children are made with CSS
+  return (
+    <div
+      className={`flex flex-col bg-container min-w-0 ${
+        open ? 'w-[450px]' : 'w-20'
+      } h-full p-2 pt-4 transition-all duration-500 max-w-[500px]`}
+    >
+      <div className="flex flex-row-reverse h-16 mb-6 justify-between items-center">
+        <div onClick={toggleOpen} className="">
```
Comment (Saelmala) [inline src/components/sideNav/SideNav.tsx:60]: The empty classname can be deleted

File: src/components/customSwitch/CustomSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <label className="inline-flex items-center cursor-pointer">
+      <input
+        type="checkbox"
+        value=""
+        className="sr-only peer"
+        onChange={onChange}
+        readOnly
+        checked={checked}
+      />
+      <div className="relative w-11 h-6 bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-green-300 dark:peer-focus:ring-green-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-[2px] after:start-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-green-600"></div>
```
Comment (Saelmala) [inline src/components/customSwitch/CustomSwitch.tsx:19]: Feels like there should be a shorter way to do this but maybe there's not

**Comment by Saelmala** [comment]
Overall LGTM, left a couple small comments. I also think that maybe the border around the toggle should not be green when the rest of the toggle isn't green either

**Review by Saelmala — APPROVED** [review]

File: src/components/sideNav/SideNav.tsx (hunk 1 above)
Comment (LucasMaupin) [reply src/components/sideNav/SideNav.tsx]: Fixed

File: src/components/customSwitch/CustomSwitch.tsx (hunk 2 above)
Comment (LucasMaupin) [reply src/components/customSwitch/CustomSwitch.tsx:19]: It has to be that long 😸

**Comment by LucasMaupin** [comment]
> Overall LGTM, left a couple small comments. I also think that maybe the border around the toggle should not be green when the rest of the toggle isn't green either
> 
> image

Fixed - it is now gray and barely visible

**Review by malmen237 — APPROVED** [review]

**Comment by malmen237** [comment]
LGTM

**Pull request merged**